### ISSUE TITLE
[Question] How would one get page locations from actions (where appropriate)?

### ISSUE THREAD

**Issue by MattAlp** [pikepdf/pikepdf#173]
Hi all, I've got the following script:
```python
from pikepdf import Pdf
path = "/some/path/on/my/machine.pdf"

with Pdf.open(path) as pdf:
    outline = pdf.open_outline()
    for title in outline.root:
        print(title)
        for subtitle in title.children:
            print('\t', subtitle)
```

which yields the following truncated output:
```
[+] The Worlds of Database Systems -> <Action>
	 [ ] The Evolution of Database Systems -> <Action>
	 [ ] Overview of a Database Management System -> <Action>
	 [ ] References -> <Action>
[+] The Relational Model of Data -> <Action>
	 [ ] An Overview of Data Models -> <Action>
	 [ ] Basics of the Relational Model -> <Action>
	 [ ] Deﬁning a Relation Schema in SQL -> <Action>
	 [ ] An Algebraic Query Language -> <Action>
	 [ ] Constraints on Relations -> <Action>
	 [ ] Summary -> <Action>
	 [ ] References -> <Action>
```
After digging around for some time, I've struggled to find a way to map these actions to page numbers. Could anyone advise or suggest other approaches to create a mapping from outline to page numbers for a PDF? Thanks.

**Comment by jbarlow83** [comment]
An item in outline can specify a simple destination, or an action. If it's a simple destination, pikepdf will resolve the page number for you. 

If it's an Action, like the PDF above, in theory the PDF can do any number of complex things like run Javascript or hide and show layers, but most of the time it will be an action that to go a destination (also easy). Or maybe a named destination (not as easy).

Then you can look up the named destination in the Names tree, which can be in multiple places, and the Names tree is a name tree, which QPDF supports but I haven't wired in yet. And after getting this, you get a page reference and position of interest on the page, and zoom level.

<img width="332" alt="Screen Shot" src="https://user-images.githubusercontent.com/1825843/109414540-35bbec80-7968-11eb-819c-d5e4571f4d00.png">

TLDR We don't handle of all the cases yet and they are legion.

**Comment by MattAlp** [comment]
That makes sense, thanks.

**Comment by mara004** [comment]
I *think* the [code](https://github.com/pikepdf/pikepdf/issues/149#issuecomment-860073511) I shared on #149 should be able to resolve a common type of actions (tested with documents produced by latexpdf).
